internal/reconstitution/api.go

```diff
@@ -99,3 +101,23 @@ func (l *lastSeenMeta) MatchesLastSeen(rv string) bool {
 	defer l.lock.Unlock()
 	return l.resourceVersion == rv
 }
+
+type lastReconciledMeta struct {
+	lock           sync.Mutex
+	lastReconciled *time.Time
+}
+
+func (l *lastReconciledMeta) ObserveReconciliation() time.Duration {
+	now := time.Now()
```

Comment:
Why not sample now after the lock? Realistically I don't think it makes a difference, just curious.

Reply:
My split second idea when writing this was to capture any time spent waiting for locks. In retrospect I am not sure how I feel about that, but practically it makes no difference since currently we don't support concurrency in the reconciliation loop.